src/card/DebitCard.java

```diff
@@ -255,7 +308,22 @@ public boolean addCard(ConnectionSource databaseConnection) throws Exception {
 				throw(new Exception("Address not added properly, abort add card"));
 			}
 		}
-
+		
+		// validate inputs
+		if (!validateCardNumber(this.getCardNumber())) {
+			throw(new Exception("Card number not in the correct format"));
+		}
+		if (!validateExpirationDate(this.getExpirationDate())) {
+			throw(new Exception("Expiration date not in the correct format"));
+		}
+		if (!validateCvv(this.getCvv())) {
+			throw(new Exception("CVV not in the correct format"));
+		}
+		if (!validatePin(this.getCardNumber())) {
+			throw(new Exception("Pin not in the correct format"));
+		}
```

Comment:
this method is pretty sizable - moving these validation checks to a validateCardInput method that could then be called here would help with readability (lines 312-324)